### PR TITLE
Refactor text into own modules, CLIPPY

### PR DESCRIPTION
**Pull request by hwchen**
Moving text functionality into their own modules for each backend.

- [x] cairo
- [x] web
- [x] svg
- [x] d2d
- [x] add docstring to text module

CLIPPY:

Sorry, probably should have done clippy in a separate PR. But there's no functional changes. Where I wasn't sure, I just put an `allowed` attribute.

- change `len() == 0` to `is_empty()`
- allow ineffective ops to have better visualization of buffer offset calculations
- allow a new without default
- in piet-common:
  - remove unneeded lifetime
  - remove redundant closure
- in piet
  - allow new without default
  - allow render context `bytes_per_pixel` pass by value instead of ref
  - allow long literals
- in d2d
  - allow most current behaviors
  - change `unwrap_or` to `unwrap_or_else`
  - allow an explicitly counted loop

### CODE REVIEW

**Comment by raphlinus** [comment]
When I run clippy locally, I get some more errors, for example:

```
error: strict comparison of f32 or f64
   --> piet-svg\src\lib.rs:291:16
    |
291 |             if width != 1.0 {
    |                ^^^^^^^^^^^^ help: consider comparing them within some error: `(width - 1.0).abs() > error`
    |
    = note: `#[deny(clippy::float_cmp)]` on by default
note: std::f32::EPSILON and std::f64::EPSILON are available.
```

File: piet/src/color.rs [hunk 1]
```diff
@@ -100,6 +100,8 @@ impl Color {
     /// which is perhaps not ideal (the clipping might change the hue). See
     /// https://github.com/d3/d3-color/issues/33 for discussion.
     #[allow(non_snake_case)]
+    #[allow(clippy::many_single_char_names)]
```
Comment (raphlinus) [inline piet/src/color.rs]: And people wonder why Rust isn't considered a first choice for scientific code :)

**Comment by hwchen** [comment]
I must have not run clippy at the workspace level :sweat_smile:

I think that I'll finish off the rest of the clippy lints in a PR that switches to github actions, since as you said it makes more sense to enforce with CI

File: piet/src/render_context.rs [hunk 2]
```diff
@@ -33,6 +33,7 @@ pub enum ImageFormat {
 }
 
 impl ImageFormat {
+    #[allow(clippy::trivially_copy_pass_by_ref)]
```
[the remark this answers is not in the document]
Comment (hwchen) [reply piet/src/render_context.rs]: Going to change this in a PR to finish off clippy lints and switch to github actions and enforce clippy.

**Pull request merged**